client/my-sites/stats/stats-video-summary/index.jsx

```diff
@@ -1,64 +1,95 @@
 /**
  * External dependencies
  */
-import React, { PropTypes } from 'react';
+import React, { Component, PropTypes } from 'react';
 import { localize } from 'i18n-calypso';
+import { connect } from 'react-redux';
+import { flowRight } from 'lodash';
 
 /**
  * Internal dependencies
  */
-import observe from 'lib/mixins/data-observe';
 import SummaryChart from '../stats-summary';
+import QuerySiteStats from 'components/data/query-site-stats';
+import { getSiteStatsNormalizedData, isRequestingSiteStatsForQuery } from 'state/stats/lists/selectors';
+import { getSelectedSiteId } from 'state/ui/selectors';
 
-const StatsVideoSummary = React.createClass( {
-	propTypes: {
-		dataList: PropTypes.object.isRequired,
-	},
-
-	mixins: [ observe( 'dataList' ) ],
+class StatsVideoSummary extends Component {
+	static propTypes = {
+		query: PropTypes.object,
+		isRequesting: PropTypes.bool,
+		siteId: PropTypes.number,
+		summaryData: PropTypes.array,
+	}
 
-	getInitialState: function() {
-		const chartDataLength = this.props.dataList.response.data ? this.props.dataList.response.data.length : null;
+	state = {
+		selectedBar: null
+	};
 
-		return {
-			selectedBar: chartDataLength ? this.props.dataList.response.data[ chartDataLength - 1 ] : null
-		};
-	},
-
-	componentWillReceiveProps: function( nextProps ) {
-		const chartDataLength = nextProps.dataList.response.data ? nextProps.dataList.response.data.length : null;
+	componentWillReceiveProps( nextProps ) {
+		const chartDataLength = nextProps.summaryData ? nextProps.summaryData.length : null;
 		// Always default to the last bar being selected
 		if ( ! this.state.selectedBar && chartDataLength ) {
 			this.setState( {
-				selectedBar: nextProps.dataList.response.data[ chartDataLength - 1 ]
+				selectedBar: nextProps.summaryData[ chartDataLength - 1 ]
 			} );
 		}
-	},
+	}
 
-	selectBar: function( bar ) {
+	selectBar = bar => {
 		this.setState( {
 			selectedBar: bar
 		} );
-	},
+	};
 
-	render: function() {
-		const { translate } = this.props;
+	render() {
+		const { query, isRequesting, moment, siteId, summaryData, translate } = this.props;
+		const data = summaryData.map( item => {
+			return {
+				...item,
+				period: moment( item.period ).format( 'MMM D' ),
+			};
+		} );
+		let selectedBar = this.state.selectedBar;
+		if ( ! selectedBar && !! data.length ) {
+			selectedBar = data[ data.length - 1 ];
```

Comment:
I thought about it first, but it's more complicated than it seems because I'm not using the `summaryData` prop directly in SummaryChart but I'm computing a derived `data` and pass it to `SummaryChart` Thus, the selectedItem won't be the same as the initial value if I initialize it on the constructor.

I could cache the `data` as well to avoid mapping, but I though that was not necessary here and adds some complexity to the component.